web/auth/auth.go

```diff
@@ -41,12 +44,37 @@ const (
 	Redirect = "redirect"
 )
 
+// Temporary solution. Will be removed when sessions replaced by JWT.
+var (
+	// Gothic http cookie sessionStore
+	sessionStore  *sessions.CookieStore
+	teacherScopes = []string{"repo:invite", "user", "repo", "delete_repo", "admin:org", "admin:org_hook"}
+	studentScopes = []string{"repo:invite"}
+	// map from session cookies to user IDs.
+	cookieStore = make(map[string]uint64)
+	httpClient  *http.Client
+)
+
+func init() {
+	httpClient = &http.Client{
+		Timeout: time.Second * 30,
+	}
+	sessionStore = sessions.NewCookieStore([]byte(rand.String()))
+	sessionStore.Options.HttpOnly = true
+	sessionStore.Options.Secure = true
+}
+
 // UserSession holds user session information.
 type UserSession struct {
 	ID        uint64
 	Providers map[string]struct{}
 }
 
+func authenticationError(logger *zap.SugaredLogger, w http.ResponseWriter, err string) {
```

Comment:
Please pass in `err error` instead of `string`. Callers should use `fmt.Errorf()` instead of `fmt.Sprintf()` or `errors.New()` if the error is simple text.